### DISCUSSION TITLE
Ongoing Maintenance

### DISCUSSION THREAD

**Discussion by MelbourneDeveloper** [MelbourneDeveloper/Device.Net#249] [Announcements]
Hello everyone,

Sorry for the long time silence on Device.Net

I've noticed some activity lately and I'd like to say that I haven't forgotten about this repo. I would really like to give it a new lease of life soon. What I'd like to do is find a single dedicated developer to maintain the codebase. Here's how that might happen...

## Funding

I think it's possible to fund someone to work on this project for a reasonable amount of time per week at a monthly rate. The person would need an understanding of low level code, USB, and HID. My new company [Nimblesite](https://www.nimblesite.co/) can provide some of that funding, but I would need a few more partners to invest in this. 

I think the funding goal would be about $500 USD per month.

This repo would then become the property of the [Nimblesite org](https://github.com/Nimblesite).

## Service Level Agreements

I would also like to offer SLAs for this library. If we can get to the funding goal of $500 per month (spread across multiple funders), Nimblesite can offer a level of support and get the main issues fixed up.

## Aims

- Fix the existing bugs
- Get the macOS and Linux versions working properly
- Get the test coverage up and test more real-life devices

If you are in a position to fund the development, or you have a lot of .NET experience working with this kind of low level code, please get in contact through [Twitter](https://x.com/CFDevelop), [LinkedIn](https://www.linkedin.com/in/christian-findlay/), or [my website](https://www.christianfindlay.com/).

But please, DO NOT send me general requests for work. If you want to work on Device.Net you absolutely must point me to similar work with USB or other low level C# code.

Christian

**Comment by domportera** [comment]
the sponsor link isn't working - it doesnt seem like sponsorship is accepted on your account atm. I can't provide *serious* funding, but I'd love to use this library someday and chip in how i can

**Reply by MelbourneDeveloper** [reply]
@domportera the sponsorship wouldn't be done through GitHub. What I'm offering is an SLA through Nimblesite for companies that are willing to provide significant funding.

**Comment by MelbourneDeveloper** [comment]
> the sponsor link isn't working - it doesnt seem like sponsorship is accepted on your account atm. I can't provide *serious* funding, but I'd love to use this library someday and chip in how i can

I removed the sponsor link a long time ago because I don't want people randomly sponsoring repos or going through the GitHub channels. Sponsorship would involve an SLA and that would be a direct agreement with the funder
